### PR TITLE
Add OAuth client

### PR DESCRIPTION
**Pull request by ottomated**
This should allow an existing google "OAuthClient" to be used with the library.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ var GoogleSpreadsheet = function( ss_key, auth_id, options ){
 
   var auth_client = new GoogleAuth();
   var jwt_client;
+  var oauth_client;
```
Comment (DerOrangeBaron) [inline index.js]: This variable is declared but its value is never used.
Please remove it.

**Comment by ScottChapman** [comment]
can we get this merged?

**Comment by ottomated** [comment]
@DerOrangeBaron any reason why it's not ready to be merged at this point?

**Comment by theoephraim** [comment]
FYI - I've just a published a new version of this module which is a complete rewrite using Google's V4 Sheets API (v3 is being deprecated on March 3, 2020). **YOU MUST MIGRATE IF YOU ARE STILL USING THIS PACKAGE!**

**Docs available here - https://theoephraim.github.io/node-google-spreadsheet**

I'll try to get this functionality merged into the new stuff ASAP.

Thanks! 😄

**Pull request closed without merging**